var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -311,6 +311,15 @@ def flag_handler(self, name, flags):
         if (name == "cflags" or name == "cxxflags") and self.spec.satisfies("%intel"):
             flags.append("-no-ipo")
             return (None, None, flags)
+
+        if name in ("cflags", "cxxflags"):
+            if "+hdf5" in self.spec:
+                if "@:5.9" in self.spec:
+                    if "@1.10:" in self.spec["hdf5"]:
+                        flags.append("-DH5_USE_18_API")
+                if "@5.10:" in self.spec:
+                    if "@1.13:" in self.spec["hdf5"]:
```

Comment:
Suggest same use of `spect.satisfies()` for version range checks at 318, 320, and 321.